src/datasets/iterable_dataset.py

```diff
@@ -82,10 +84,94 @@ def hasnext(self):
         return self._hasnext
 
 
+def _convert_to_arrow(
+    iterable: Iterable[Tuple[Key, dict]],
+    batch_size: int,
+    drop_last_batch=False,
+) -> Iterator[Tuple[Key, pa.Table]]:
+    """Iterate over sub-tables of size `batch_size`.
```

Comment:
The definition in the docstring is the same as the one for `_batch_arrow_tables` below. Maybe we should add to the description they expect different iterables...

Maybe also the naming of both functions could be aligned: now they have very different names whereas their functionality is analogous...

Reply:
I like the idea of having different names because one is expensive (contains "convert" in the name) while the other one is not (simply batches data)